core/src/main/java/io/micronaut/core/naming/NameUtils.java

```diff
@@ -214,6 +215,41 @@ public static String getSimpleName(String className) {
         return className;
     }
 
+    /**
+     * Returns the shortened fully-qualified name for a class represented as a string.
+     * Shortened name would have package names and owner objects reduced to a single letter.
+     * For example, {@code com.example.Owner$Inner} would become {@code c.e.O$Inner}.
+     * IDEs would still be able to recognize these types, but they would take less space
+     * visually.
+     *
+     * @since 4.8.x
+     * @param typeName The fully-qualified type name
+     * @return The shortened type name
+     */
+    @Experimental
+    public static String getShortenedName(String typeName) {
+        int nameStart = typeName.lastIndexOf('$');
```

Comment:
check type name doesn't == null

Reply:
That is not checked in other methods, so I assume it is not typical for `typeName` to be `null`. Added `@NonNull` though.